js/index.js

```diff
@@ -38,5 +38,57 @@ const siteContent = {
 };
 
 // Example: Update the img src for the logo
-let logo = document.getElementById("logo-img");
-logo.setAttribute('src', siteContent["nav"]["img-src"])
+// let logo = document.getElementById("logo-img");
+// logo.setAttribute('src', siteContent["nav"]["img-src"]);
+
+//nav
+
+const nav = document.querySelector('nav');
+console.log(nav.querySelectorAll('a'));
+
+nav.querySelectorAll('a')[0].textContent = siteContent.nav["nav-item-1"];
+nav.querySelectorAll('a')[1].textContent = siteContent.nav["nav-item-2"];
+nav.querySelectorAll('a')[2].textContent = siteContent.nav["nav-item-3"];
+nav.querySelectorAll('a')[3].textContent = siteContent.nav["nav-item-4"];
+nav.querySelectorAll('a')[4].textContent = siteContent.nav["nav-item-5"];
+nav.querySelectorAll('a')[5].textContent = siteContent.nav["nav-item-6"];
+
+// cta
+
+document.querySelector('#cta-img').src = siteContent.cta["img-src"];
+document.querySelector('.cta .cta-text h1').textContent = siteContent.cta["h1"];
+document.querySelector('.cta .cta-text button').textContent = siteContent.cta["button"];
+
+let main = document.querySelector('.main-content');
+console.log(main.querySelector('.top-content .text-content'));
+
+// h4 elements for top-content
+main.querySelectorAll('.top-content .text-content')[0].querySelector('h4').textContent = siteContent["main-content"]["features-h4"];
+main.querySelectorAll('.top-content .text-content')[1].querySelector('h4').textContent = siteContent["main-content"]["about-h4"];
+
+// p elements for top-content
+main.querySelectorAll('.top-content .text-content')[0].querySelector('p').textContent = siteContent["main-content"]["features-content"];
+main.querySelectorAll('.top-content .text-content')[1].querySelector('p').textContent = siteContent["main-content"]["about-content"];
+
+// main img
+main.querySelector('.middle-img').src = siteContent["main-content"]["middle-img-src"];
+
+// h4 elements for bottom-content
+main.querySelectorAll('.bottom-content .text-content')[0].querySelector('h4').textContent = siteContent["main-content"]["services-h4"];
```

Comment:
You should always try to store the values of your selectors in a temporary variable so you don't have to run the calculation multiple times to get the same elements. You could for instance save the `text-content` values like so:
```js
const mainContentH4s = main.getElementsByTagName('h4');
```
So, for all these values, you're making a lot of assignments. How could we make it more DRY so we're not repeating code?

Well, I see another opportunity for loops here! We just need to make sure that we're referring to the correct `siteContent` values for each `-h4` and `-content` key. If we check out `siteContent`, we can see that the order in which our keys show up are the same as how they are presented on the DOM.

So, we could just grab the h4 items in an array and map the indexes to h4 nodes we gathered.

For example:
```js
// first, we grab all the `siteContent` keys into an array
const siteContentKeys = Object.keys(siteContent['main-content']);
// now we need to grab the h4 keys specifically - luckily every h4 element ends with the string 'h4'
const siteContentH4s = siteContentKeys.filter(key => key.endsWith('h4'));
// and finally, we can loop through our h4 elements and set them using the equivalent index of our `mainContentH4s` array
siteContentH4s.forEach((value, idx) => {
  mainContentH4s[idx].textContent = siteContent['main-content'][value];
});
```